File: src/lib/api/session2mdSettings.ts
import { invoke } from "@tauri-apps/api/core";
import {
  SESSION_DIRECTORY_IDS,
  type SessionDirectoryId,
} from "@/lib/sessionProviders";

export { SESSION_DIRECTORY_IDS };
export type { SessionDirectoryId };

export interface Session2mdSettings {
  directoryOverrides: Partial<Record<SessionDirectoryId, string>>;
}

export interface Session2mdSettingsSnapshot extends Session2mdSettings {
  resolvedDirectories: Record<SessionDirectoryId, string>;
}

export const session2mdSettingsApi = {
  get: (): Promise<Session2mdSettingsSnapshot> =>
    invoke("get_session2md_settings"),

  save: (settings: Session2mdSettings): Promise<Session2mdSettingsSnapshot> =>
    invoke("save_session2md_settings", { settings }),
};

File: src-tauri/src/commands/session2md_settings.rs
use crate::session2md_settings::{self, Session2mdSettings, Session2mdSettingsSnapshot};

#[tauri::command]
pub fn get_session2md_settings() -> Result<Session2mdSettingsSnapshot, String> {
    session2md_settings::get_settings_snapshot()
}

#[tauri::command]
pub fn save_session2md_settings(
    settings: Session2mdSettings,
) -> Result<Session2mdSettingsSnapshot, String> {
    session2md_settings::save_settings(settings)
}
